internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -307,7 +449,7 @@ func (f *fleetGateway) convertToCheckinComponents(components []runtime.Component
 	return checkinComponents
 }
 
-func (f *fleetGateway) execute(ctx context.Context) (*fleetapi.CheckinResponse, time.Duration, error) {
+func (f *fleetGateway) execute(ctx context.Context, state state.State) (*fleetapi.CheckinResponse, time.Duration, error) {
```

Comment:
You're going to need to change the `CheckinRequest` to add a `poll_timeout` string attribute.